stratum/hal/lib/barefoot/bfrt_table_manager.cc

```diff
@@ -102,6 +133,71 @@ ::util::Status BfrtTableManager::VerifyForwardingPipelineConfig(
   return ::util::OkStatus();
 }
 
+::util::Status BfrtTableManager::HandleRegisterReset(
+    std::vector<p4::config::v1::Register>& registers) {
+  LOG(INFO) << "Started reset thread for registers.";
+  auto session = bfrt::BfRtSession::sessionCreate();
+  CHECK_RETURN_IF_FALSE(session != nullptr) << "Unable to create session.";
+  ::util::Status status = ::util::OkStatus();
+  for (;;) {
+    auto t1 = absl::Now();
+    uint64 min_reset_interval_ms = 10000;
+    RETURN_IF_BFRT_ERROR(session->beginBatch());
+    for (const auto& reg : registers) {
+      P4Annotation annotation;
+      {
+        absl::ReaderMutexLock l(&lock_);
+        ASSIGN_OR_RETURN(
+            annotation,
+            p4_info_manager_->GetSwitchStackAnnotations(reg.preamble().name()));
+      }
+      std::string clear_value =
+          Uint64ToByteStream(annotation.register_reset_value());
+      min_reset_interval_ms = std::min(annotation.register_reset_interval_ms(),
+                                       min_reset_interval_ms);
+      ::p4::v1::RegisterEntry register_entry;
+      register_entry.set_register_id(reg.preamble().id());
+      register_entry.mutable_data()->set_bitstring(clear_value);
+      register_entry.clear_index();
+      // TODO(max): remove after testing
+      if (mode_ == OPERATION_MODE_SIM) {
+        register_entry.mutable_index()->set_index(0);
+      }
+      ::util::Status status =
```

Comment:
I guess you want to append the status from `WriteRegisterEntry` to the outer status(line141)?